### PR TITLE
Use package_json gem

### PR DESCRIPTION
**Pull request by vaukalak**
### Summary

Since shakapacker V8+ uses package_json gem, this is now supported in react_on_rails as well. This PR closes [issue #1635.](https://github.com/shakacode/react_on_rails/issues/1635).

### Pull Request checklist
_Remove this line after checking all the items here. If the item is not applicable to the PR, both check it out and wrap it by `~`._

- [x] Add/update test to cover these changes
- [x] Update documentation
- [x] Update CHANGELOG file  
  _Add the CHANGELOG entry at the top of the file._

- - -
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/1639)

## Summary by CodeRabbit

- **Improvements**
  - Enhanced dependency management for easier and more reliable package handling.
  - Updated installation documentation for a streamlined setup process.
  - Improved support for modern versions of Shakapacker.

- **Documentation**
  - Simplified installation instructions in the Getting Started guide.

- **Development**
  - Added the `package_json` gem to development dependencies for better package management.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -119,6 +119,9 @@ GEM
       tins (~> 1.6)
     crass (1.0.6)
     date (3.3.4)
+    debug (1.9.2)
```
Comment (vaukalak) [inline Gemfile.lock]: I think those deps are introduced by package_json

File: lib/generators/react_on_rails/generator_helper.rb [hunk 2]
```diff
@@ -1,8 +1,13 @@
 # frozen_string_literal: true
 
+require "package_json"
 require "rainbow"
 
 module GeneratorHelper
+  def package_json
+    @package_json ||= PackageJson.read(Rails.root.to_s)
```
Comment (vaukalak) [inline lib/generators/react_on_rails/generator_helper.rb]: @G-Rath I've changed it vs how it was implemented [here](https://github.com/G-Rath/react_on_rails/commit/6c0a29ebff3b584b6446f1065dec23b6bf7ffb50#diff-84d1b667e35dc8d3cf92e40813c3b880c84a1d31dbd6d6367e8851f84418b435R6) cause it was creating a separate bundle context, due to which I couldn't locate the shakapacker gem during the install generator work.

File: Gemfile.lock (hunk 1 above)
Comment (G-Rath) [reply Gemfile.lock]: nope, `package_json` has no dependencies outside of core Ruby, and it's not being added as an explicitly dependency anyway (which I think is fine so long as it's only used in the generator - that's why Shakapacker _does_ directly depend on it, but react-rails doesn't)

**Review by G-Rath — APPROVED** [review]

File: lib/generators/react_on_rails/react_with_redux_generator.rb [hunk 3]
```diff
@@ -5,6 +5,7 @@
 module ReactOnRails
   module Generators
     class ReactWithReduxGenerator < Rails::Generators::Base
+      include GeneratorHelper
```
Comment (justin808) [inline lib/generators/react_on_rails/react_with_redux_generator.rb]: maybe not necessary?

**Review by justin808 — CHANGES_REQUESTED** [review]

**Pull request merged**